firmware/Sources/MID/MID_DABS/mid_dabs.c

```diff
@@ -0,0 +1,292 @@
+ * @return @ref MID_DABS_RESULT_SUCCESS if measured correctly,
+ * 		@ref MID_DABS_RESULT_BUSY, @ref MID_DABS_RESULT_TIMEOUT or
+ * 		@ref MID_DABS_RESULT_ERROR otherwise.
+ */
+static MID_DABS_result_e SetLeds(uint8_t step){
+	MID_DABS_result_e res = MID_DABS_RESULT_ERROR;
+	uint8_t state = ledsMode.mode.ledsStep[step];
+	uint8_t mask= 0x1;
+	res = (MID_DABS_result_e) HAL_GpioSet(HAL_GPIO_OUT_Led0, state & mask);
+	res = (MID_DABS_result_e) HAL_GpioSet(HAL_GPIO_OUT_Led1, (state>>1) & mask);
+	res = (MID_DABS_result_e) HAL_GpioSet(HAL_GPIO_OUT_Led2, (state>>2) & mask);
+	res = (MID_DABS_result_e) HAL_GpioSet(HAL_GPIO_OUT_Led3, (state>>3) & mask);
+	return res;
+}
+
+/**********************************************************************************/
+/*                        Definition of exported functions                        */
+/**********************************************************************************/
+
+//MID_DABS_result_e MID_DabsUpdateMeas(const MID_DABS_meas_e type, MID_REG_meas_s * measreg){
+//	MID_DABS_result_e res = MID_DABS_RESULT_SUCCESS;
+//	switch(type){
+//		case MID_DABS_MEAS_ELECTRIC:
+//			uint16_t data;
+//			res = (MID_DABS_result_e) HAL_StsReadTemperature(&temp);
+//			if (res == MID_DABS_RESULT_SUCCESS){
+//				*measreg->hsVolt = data;
+//			}
+//			res = (MID_DABS_result_e) HAL_StsReadTemperature(&temp);
+//			if (res == MID_DABS_RESULT_SUCCESS){
+//				*measreg->tempBody = data;
+//			}
+//			int16_t data;
+//			res = (MID_DABS_result_e) HAL_StsReadTemperature(&temp);
+//			if (res == MID_DABS_RESULT_SUCCESS){
+//				*measreg->tempBody = data;
+//			}
+//			break;
+//		case MID_DABS_MEAS_TEMP:
+//			int16_t temp;
+//			// Check if the hardware version has I2C
+//			if (MID_REG_info_s.hwVer > 16){
+//				res = (MID_DABS_result_e) HAL_StsReadTemperature(&temp);
+//				if (res == MID_DABS_RESULT_SUCCESS){
+//					*measreg->tempBody = temp;
+//				}
+//			}else{
+//				// If not I2C the register will always have a 0
+//				*measreg->tempBody=0;
+//			}
+//			res = (MID_DABS_result_e) HAL_AdcGet(HAL_ADC_PORT_Ext_Temp1, &temp);
+//			if (res == MID_DABS_RESULT_SUCCESS){
+//				*measreg->tempBody = temp;
+//			}
+//			res = (MID_DABS_result_e) HAL_AdcGet(HAL_ADC_PORT_Ext_Temp2, &temp);
+//			if (res == MID_DABS_RESULT_SUCCESS){
+//				*measreg->tempBody = temp;
+//			}
+//			break;
+//	}
+//	return res;
+//}
+
+
+MID_DABS_result_e MID_DabsUpdateLeds(MID_REG_mode_e ctrlMode,
+		int16_t curr, MID_REG_errorStatus_s * errors){
+	MID_DABS_result_e res = MID_DABS_RESULT_ERROR;
+	blink_conf_s inputMode;
+	//Check the mode of the EPC and the mode the leds should be in
+	res = CheckBlinkStatus(ctrlMode, curr, errors, &inputMode);
+	if (res == MID_DABS_RESULT_SUCCESS){
+		if (ledsMode.prevMode.mode == inputMode.mode){
+			ledsMode.step++;
+			ledsMode.step = ledsMode.step % ledsMode.mode.steps;
+		}
+		else{
+			ledsMode.mode = inputMode;
+			ledsMode.step = 0;
+		}
+		res = SetLeds(ledsMode.step);
```

Comment:
you're setting leds even if they have only 1 step

Reply:
Solved